lib/helpers.rb

```diff
@@ -19,6 +19,14 @@ def comment
     @comment ||= Comment.find(params[:comment_id])
   end
 
+  def verify_or_fix_cached_comment_count(comment, comment_hash)
+    unless comment_hash["children"].nil?
+      if comment_hash["child_count"] != comment_hash["children"].length
+        comment_hash["child_count"] = comment.update_cached_child_count
```

Comment:
This would then need to be updated to explicitly call `comment.update_cached_child_count`, and then `comment.get_cached_child_count`.